### ISSUE TITLE
Build issue with ITK on Apple Silicon

### ISSUE THREAD

**Issue by dcoeurjo** [DGtal-team/DGtal#1619]
When build DGtal on a M1, there is no problem except when setting `-DWITH_ITK=true` which produces:

``` 
Consolidate compiler generated dependencies of target DGtal
[  1%] Building CXX object CMakeFiles/DGtal.dir/src/DGtal/base/Bits.cpp.o
In file included from /Users/davidcoeurjolly/Sources/DGtal/src/DGtal/base/Bits.cpp:30:
In file included from /Users/davidcoeurjolly/Sources/DGtal/src/DGtal/base/Bits.h:32:
In file included from /Users/davidcoeurjolly/Sources/DGtal/src/DGtal/base/Common.h:99:
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.0.sdk/usr/include/c++/v1/cmath:321:9: error: no member named 'signbit' in the global namespace
using ::signbit;
      ~~^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.0.sdk/usr/include/c++/v1/cmath:322:9: error: no member named 'fpclassify' in the global namespace
using ::fpclassify;
      ~~^
...
```
Ping @kerautret

**Comment by dcoeurjo** [comment]
Have you already epxerienced something like that @kerautret ?

**Comment by kerautret** [comment]
yes I see it on the GitHub Actions when trying to activate ITK on MacOS build, it looks like that the include of cmath.h  from Xcode build system find first the math.h of /usr/local/include before the one of Xcode Tools chain.

**Comment by kerautret** [comment]
Same problem here trying to activate ITK and macos GitHub Actions: (with a cmake small try fix)
https://github.com/kerautret/DGtal/runs/4445372646?check_suite_focus=true

**Comment by kerautret** [comment]
@dcoeurjo have you tried to use the the CommandLineTools SDK instead the XCode SDK, by using: 
`sudo xcode-select -s /Library/Developer/CommandLineTools` 
It looks to be better when using ITK from brew (I tested on my non M1 Mac it works while also failing with Xcode SDK)

**Comment by dcoeurjo** [comment]
THanks for the tip, I'm not super fan of changing the SDK globally.. (I wasn't expecting that the command line has a different sdk than Xcode.. weird..)

**Comment by kerautret** [comment]
> THanks for the tip, I'm not super fan of changing the SDK globally.. (I wasn't expecting that the command line has a different sdk than Xcode.. weird..)
I am agree not ideal, for instance I detected that Unreal Engine was no more working by using this SDK! 😱

**Comment by kerautret** [comment]
@dcoeurjo I tested a simple ITK helloworld  project and it fails from the same errors so the problem is not related to DGtal itself but more from brew/ITK/osXSDK. I try to ask it on the ITK dev forum if they get a good workaround to fix it :).
